expo/expo
# fix(config-plugins): default to `['dangerous']` sort order when using unknown platforms

# Why

Follow-up of #22201

# How

When using arbitrary platforms, like `foobar`, the sort order is not defined. This results in errors like:

![image](https://user-images.githubusercontent.com/1203991/233685476-24afd065-77a5-4f39-9d09-a49c20dd7b4f.png)

Instead, this change adds a default sorting order of `['dangerous']` when using arbitrary platforms.

# Test Plan

- Clone the example repository: https://github.com/byCedric/custom-prebuild-example
- Create a tarball from template `$ cd ./template && npm pack`
- Run `$ expo prebuild --platform macos --template ../template/custom-prebuild-template-1.0.0.tgz`

This should not fail as the screenshot above.

# Checklist
